src/marlin_stubs/M865.cpp

```diff
@@ -37,12 +38,8 @@ void PrusaGcodeSuite::M865() {
 
     FilamentType filament_type;
 
-    if (const auto slot = p.option<uint8_t>('I', static_cast<uint8_t>(0), static_cast<uint8_t>(adhoc_filament_type_count - 1))) {
-        filament_type = AdHocFilamentType { .tool = *slot };
-        if (config_store().get_filament_type(*slot) != filament_type) {
-            SERIAL_ERROR_MSG("The selected tool does not have the ad-hoc filament loaded. Changes will have no effect.");
-        }
-
+    if (const auto slot = p.option<uint8_t>('I', static_cast<uint8_t>(0), static_cast<uint8_t>(EXTRUDERS - 1))) {
```

Comment:
Changing behavior (and documentation) of the I parameter should be a separate commit